plugins/generic/htmlArticleGalley/HtmlArticleGalleyPlugin.inc.php

```diff
@@ -93,53 +96,26 @@ function headerCallback($hookName, $args) {
 					}
 				}
 			}
-		}
-	}
-
-	/**
-	 * @see ViewableFilePlugin::displayArticleGalley
-	 */
-	function displayArticleGalley($templateMgr, $request, $params) {
-		$journal = $request->getJournal();
-
-		if (!$journal) return '';
-		$fileId = (isset($params['fileId']) && is_numeric($params['fileId'])) ? (int) $params['fileId'] : null;
-
-		$galley = $templateMgr->get_template_vars('galley'); // set in ArticleHandler
-		$templateMgr->assign('htmlGalleyContents', $this->_getHTMLContents($request, $galley, $fileId));
-		return parent::displayArticleGalley($templateMgr, $request, $params);
+		} */
 	}
 
 	/**
 	 * Return string containing the contents of the HTML file.
 	 * This function performs any necessary filtering, like image URL replacement.
-	 * @param $fileId int optional file id, otherwise the 'best' file will be chosen.
+	 * @param $request PKPRequest
+	 * @param $galley ArticleGalley
 	 * @return string
 	 */
-	function _getHTMLContents($request, $galley, $fileId = null) {
+	function _getHTMLContents($request, $galley) {
 		import('lib.pkp.classes.file.SubmissionFileManager');
 		$fileManager = new SubmissionFileManager($request->getContext()->getId(), $galley->getSubmissionId());
-		if (!$fileId) {
-
-			// Note: Some HTML file uploads may be stored with incorrect file_type settings
-			// due to incorrect finfo or mime.magic entries.  As such, we examine the file extension
-			// of the original file name for 'htm'. This will match .html, .htm, .xhtml, etc.
-			// The file_type isn't important since the plugin includes the HTML content inline rather
-			// than including a URL loaded in an iframe.
-			$file = $galley->getFirstGalleyFile('htm');
-			if ($file) {
-				$fileId = $file->getFileId();
-			} else {
-				assert(false); // No HTML file in this HTML Article galley?
-				return false;
-			}
-		}
-		$contents = $fileManager->readFile($fileId);
+		$contents = $fileManager->readFile($galley->getFile()->getFileId());
+print_r($galley->getFile()->getFileId());
```

Comment:
This plugin still needs a lot of rewriting -- I'll add it to the list and update the issue entry. I'm unlikely to get to this over the next few days. It'll need e.g. was to associate image files and CSS files with a galley, which none of the other formats have required.